src/map/skill.c

```diff
@@ -10938,6 +10938,32 @@ static int skill_count_wos(struct block_list *bl, va_list ap)
 	return 0;
 }
 
+/**
+ * Returns the ID of the linked song/dance (Bard/Dancer Soul Link)
+ */
+static int skill_get_linked_song_dance_id(int skill_id)
+{
+	switch (skill_id) {
+		case BA_WHISTLE:
+			return DC_HUMMING;
+		case BA_ASSASSINCROSS:
+			return DC_DONTFORGETME;
+		case BA_POEMBRAGI:
+			return DC_FORTUNEKISS;
+		case BA_APPLEIDUN:
+			return DC_SERVICEFORYOU;
+		case DC_HUMMING:
+			return BA_WHISTLE;
+		case DC_DONTFORGETME:
+			return BA_ASSASSINCROSS;
+		case DC_FORTUNEKISS:
+			return BA_POEMBRAGI;
+		case DC_SERVICEFORYOU:
+			return BA_APPLEIDUN;
+	}
```

Comment:
in here it silently fail, should not assert instead?

Reply:
I'll document that it returns 0 if there's no linked ID, so that it can be used in a place that doesn't need any knowledge of the skill IDs if we need to

Reply:
Done